apps/src/calc/token.js

```diff
@@ -0,0 +1,95 @@
+var jsnums = require('./js-numbers/js-numbers');
+
+/**
+ * A token is a value, and a boolean indicating whether or not it is "marked".
+ * Marking is done for two different reasons.
+ * (1) We're comparing two expressions and want to mark where they differ.
+ * (2) We're looking at a single expression and want to mark the deepest
+ *     subexpression.
+ * @param {} val
+ * @param {boolean} marked
+ */
+var Token = function (val, marked) {
+  this.val_ = val;
+  this.marked_ = marked;
+
+  // Store string representation of value. In most cases this is just a
+  // non repeated portion. In the case of something like 1/9 there will be both
+  // a non repeated portion "0." and a repeated portion "1" - i.e. 0.1111111...
+  this.nonRepeated_ = null;
+  this.repeated_ = null;
+  this.setStringRepresentation_();
+};
+module.exports = Token;
+
+/**
+ * Add the given token to the parent element.
+ * @param {HTMLElement} element Parent element to add to
+ * @param {number} xPos X position to place element at
+ * @param {string?} markClass Class name to use if token is marked
+ * @returns {number} the length of hte added text element
+ */
+Token.prototype.addToParent = function (element, xPos, markClass) {
```

Comment:
Do these get manipulated after being added? If not, something like `renderToParent` could express the view-iness of it